src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1845,11 +1845,10 @@ protected override void InvokeDefaultAction(string path)
                                 //  a) the user has asked to with the -FollowSymLinks switch parameter and
```

Comment:
@SteveL-MSFT, your last commit had 4 failures in `PowerShell-CI-windows`
Hard link and symbolic link tests.New-Item and hard/symbolic links.New-Item can create symbolic link to file
```powershell
Expected strings to be the same, but they were different.
Expected length: 90
Actual length:   94
Strings differ at index 0.
Expected: 'C:\Users\V...'
But was:  '\??\C:\Use...'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystem.Tests.ps1: line 478
478:             $link.Target | Should -BeExactly $real.ToString()
```
Hard link and symbolic link tests.New-Item and hard/symbolic links.New-Item can create a symbolic link to nothing
```powershell
Expected strings to be the same, but they were different.
Expected length: 92
Actual length:   96
Strings differ at index 0.
Expected: 'C:\Users\V...'
But was:  '\??\C:\Use...'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystem.Tests.ps1: line 486
486:             $link.Target | Should -Be $nonFile.ToString()
```
Hard link and symbolic link tests.New-Item and hard/symbolic links.New-Item can create a directory symbolic link to a directory
```powershell
Expected strings to be the same, but they were different.
Expected length: 88
Actual length:   92
Strings differ at index 0.
Expected: 'C:\Users\V...'
But was:  '\??\C:\Use...'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\FileSystem.Tests.ps1: line 506
506:             $link.Target | Should -BeExactly $real.ToString()
```
New-Item with links.Should create a symbolic link to a non-existing file without error
```powershell
Expected strings to be the same, but they were different.
Expected length: 98
Actual length:   102
Strings differ at index 0.
Expected: 'C:\Users\V...'
But was:  '\??\C:\Use...'
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Management\New-Item.Tests.ps1: line 228
228:         $fileInfo.Target | Should -Be $target
```